src/plugins/immutable-properties.ts

```diff
@@ -209,11 +230,15 @@ export function createPlugin():ValueObject.Plugin {
       return Maybe.Nothing<Code.FileType>();
     },
     forwardDeclarations: function(valueType:ValueObject.Type):ObjC.ForwardDeclaration[] {
+      const makePublicImports = makePublicImportsForValueType(valueType);
       const typeLookupForwardDeclarations = valueType.typeLookups.filter(FunctionUtils.pApplyf2(valueType, isForwardDeclarationRequiredForTypeLookup))
                                                                  .map(forwardDeclarationForTypeLookup);
-      const attributeForwardDeclarations = valueType.attributes.filter(FunctionUtils.pApply2f3(valueType.typeName, makePublicImportsForValueType(valueType), shouldForwardDeclareAttribute))
-                                                               .map(forwardDeclarationForAttribute);
-      return [].concat(typeLookupForwardDeclarations).concat(attributeForwardDeclarations);
+      const attributeForwardClassDeclarations = valueType.attributes.filter(FunctionUtils.pApply2f3(valueType.typeName, makePublicImports, shouldForwardDeclareAttribute))
+                                                               .map(forwardClassDeclarationForAttribute);
+      const attributeForwardProtocolDeclarations = makePublicImports
+        ? []
+        : valueType.attributes.map(forwardProtocolDeclarationForAttribute).filter(function(v) { return v !== undefined });
```

Comment:
This should be filtered ahead of time so we're not filtering out undefined after the map